### PR TITLE
Simpler `#halt` testing semantics, explicit state-splitting on `JUMPI` instruction

### PR DESCRIPTION
**Pull request by ehildenb**

### CODE REVIEW

File: tests/templates/output-success-llvm.json [hunk 1]
```diff
@@ -0,0 +1,162 @@
+<kevm>
```
Comment (dwightguth) [inline tests/templates/output-success-llvm.json]: why is this file here? it shouldn't actually be using krun when running the llvm backend, right?

Comment (ehildenb) [reply tests/templates/output-success-llvm.json]: The `./kevm interpret ` command does `k-bin-to-text` on the output if it sees a non-zero exit code, and then the `Makefile` does `diff` on that and the backend-specific output file.

Comment (ehildenb) [reply tests/templates/output-success-llvm.json]: Note that both `k-bin-to-text` and `diff` are skipped on zero exit code.

Comment (ehildenb) [reply tests/templates/output-success-llvm.json]: We haven't needed the output file until now, because I haven't introduced bugs which we needed to display the diff for. Only on this PR (which introduced a bug) did the file ommission show up.

Comment (ehildenb) [reply tests/templates/output-success-llvm.json]: This was resolved in #380 .
